src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,148 @@
+package duke;
+
+import command.UserInterface;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+
+/**
+ * Parser class will parse the input data and generate a taskList from the input the file.
+ */
+public class Parser {
+    /**
+     * parseCommand method reads the tasks in the input file and generates a taskList so that the chatbot
+     * can load the tasks from input file.
+     *
+     * @param command       the command
+     * @param taskManager   the task manager
+     * @param userInterface the user interface
+     */
+    public static String parseCommand(String command, TaskManager taskManager, UserInterface userInterface, Storage storage) {
+        String[] parts = command.split(" ", 2);
+        String action = parts[0];
+        StringBuilder output = new StringBuilder("");
+
+        switch (action) {
+        case "todo":

```

Comment:
Perhaps you would want to abstract this into several classes so that this file does not get too big and makes it easier to understand.